pkg/simple/client/application/helper.go

```diff
@@ -571,14 +571,16 @@ func GenerateShortNameMD5Hash(input string) string {
 
 func FormatVersion(input string) string {
 	re := regexp.MustCompile(`[^a-z0-9-.]`)
-	errs := validation.IsDNS1123Subdomain(input)
-	if len(errs) != 0 {
-		klog.Warningf("Version %s does not meet the Kubernetes naming standard, replacing invalid characters with '-'", input)
-		input = re.ReplaceAllStringFunc(input, func(s string) string {
-			return "-"
-		})
-	}
-	return input
+	if len(validation.IsDNS1123Subdomain(input)) == 0 {
+		return input
+	}
+	invalidCharsStr := strings.Join(re.FindAllString(input, -1), "")
+	hash := md5.Sum([]byte(invalidCharsStr))
```

Comment:
Although security is usually not the primary concern when generating Kubernetes resource names, MD5 has known weaknesses and is susceptible to collision attacks (i.e., generating the same MD5 hash for different inputs). SHA-1 and SHA-256 are more collision-resistant, which is why MD5 is generally not recommended in most modern application scenarios.